app/models/user.rb

```diff
@@ -317,13 +317,18 @@ def self.find_or_create_external_user(attrs, auth_source_name)
     else
       User.as_anonymous_admin do
         auth_source = AuthSourceExternal.create!(:name => auth_source_name) if auth_source.nil?
-        user = User.create!(attrs.merge(:auth_source => auth_source))
-        user.locations = auth_source.locations
-        user.organizations = auth_source.organizations
-        if external_groups.present?
-          user.usergroups = auth_source.external_usergroups.where(:name => external_groups).map(&:usergroup).uniq
+        user = User.new(attrs.merge(:auth_source => auth_source))
+        if user.save
+          user.locations = auth_source.locations
+          user.organizations = auth_source.organizations
+          if external_groups.present?
+            user.usergroups = auth_source.external_usergroups.where(:name => external_groups).map(&:usergroup).uniq
+          end
+          user.post_successful_login
+        else
+          logger.info "Unable to perform external user login: #{user.errors.full_messages.join(', ')}"
```

Comment:
I'd suggest to be more specific, the creation failed, thus the login failed.
```suggestion
          logger.info "Failed to create external User '{user.login}' : #{user.errors.full_messages.join(', ')}"
```